src/transforms/pipelines/mod.rs

```diff
@@ -0,0 +1,200 @@
+
+    fn names(&self) -> Vec<String> {
+        if let Some(ref names) = self.order {
+            names.clone()
+        } else {
+            let mut names = self.pipelines.keys().cloned().collect::<Vec<String>>();
+            names.sort();
+            names
+        }
+    }
+
+    fn serial(&self) -> Box<dyn TransformConfig> {
+        let pipelines: IndexMap<String, Box<dyn TransformConfig>> = self
+            .names()
+            .into_iter()
+            .filter_map(|name| {
+                self.pipelines
+                    .get(&name)
+                    .map(|config| (name, config.serial()))
+            })
+            .collect();
+
+        Box::new(expander::ExpanderConfig::serial(pipelines))
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+// #[serde(deny_unknown_fields)]
+pub struct PipelinesConfig {
+    #[serde(default)]
+    logs: EventTypeConfig,
+    #[serde(default)]
+    metrics: EventTypeConfig,
+}
+
+impl PipelinesConfig {
+    fn parallel(&self) -> IndexMap<String, Box<dyn TransformConfig>> {
+        let mut map: IndexMap<String, Box<dyn TransformConfig>> = IndexMap::new();
+
+        if !self.logs.is_empty() {
+            map.insert(
+                "logs".to_string(),
+                Box::new(router::EventRouterConfig::log(self.logs.serial())),
+            );
+        }
+
+        if !self.metrics.is_empty() {
+            map.insert(
+                "metrics".to_string(),
+                Box::new(router::EventRouterConfig::metric(self.metrics.serial())),
+            );
+        }
+
+        map
+    }
+}
+
+#[async_trait::async_trait]
+#[typetag::serde(name = "pipelines")]
+impl TransformConfig for PipelinesConfig {
+    async fn build(&self, _context: &TransformContext) -> crate::Result<Transform> {
+        Err("this transform must be expanded".into())
+    }
+
+    fn expand(
+        &mut self,
+    ) -> crate::Result<Option<(IndexMap<String, Box<dyn TransformConfig>>, ExpandType)>> {
+        Ok(Some((
+            self.parallel(),
+            ExpandType::Parallel { aggregates: true },
+        )))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Any
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Any
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "pipelines"
+    }
+}
+
+impl GenerateConfig for PipelinesConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(SAMPLE_CONFIG).unwrap()
+    }
+}
+
+#[cfg(test)]
+impl PipelinesConfig {
+    pub fn from_toml(input: &str) -> Self {
+        crate::config::format::deserialize(input, Some(crate::config::format::Format::Toml))
+            .unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{PipelinesConfig, SAMPLE_CONFIG};
+
+    #[test]
+    fn generate_config() {
+        crate::test_util::test_generate_config::<PipelinesConfig>();
+    }
+
+    #[test]
+    fn parsing() {
+        let config = PipelinesConfig::from_toml(SAMPLE_CONFIG);
```

Comment:
Take it or leave it: seems like you might want to just move the `indoc!` call to be inlined in `PipelinesConfig::generate_config`, similar to how it's done [here](https://github.com/vectordotdev/vector/blob/b7e1dfcb21506d58777cd2b0dcb44ae451ed47b2/src/sinks/humio/metrics.rs#L58-L67), and then `PipelinesConfig::from_toml` can just take the `toml::Value` directly.